### PR TITLE
[FEAT] CCIP Wildcard Support

### PR DESCRIPTION
**Pull request by gantunesr**
**Description**

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions,_
_1. What is the reason for the change?_
(This was originally a [community PR](https://github.com/MetaMask/metamask-mobile/pull/4322).)

Support for https://github.com/MetaMask/specifications/issues/9
- Relates to https://github.com/MetaMask/metamask-extension/pull/14675

_2. What is the improvement/solution?_
Add new ENS provider for Wildcard resolutions and introduce the package `@ethersproject/shims` to address RN network limitations.

For more information about `@ethersproject/shims` use the following resources,

- https://docs.ethers.io/v5/cookbook/react-native/
- https://github.com/ethers-io/ethers.js/tree/master/packages/shims
- https://snyk.io/advisor/npm-package/@ethersproject/shims

**Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [x] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Review by Fatxx — APPROVED** [review]
LGTM

File: package.json [hunk 1]
```diff
@@ -283,6 +284,7 @@
     "@types/react-native-vector-icons": "^6.4.8",
     "@types/react-native-video": "^5.0.13",
     "@types/redux-mock-store": "^1.0.3",
+    "@types/sinon": "^10.0.11",
```
Comment (blackdevelopa) [inline package.json:287]: I don't seem to see where we use this `@types/sinon` in the PR.

**Review by blackdevelopa — COMMENTED** [review]
LGTM

File: package.json (hunk 1 above)
Comment (gantunesr) [reply package.json]: `@types/sinon` lives in your heart

Removed in https://github.com/MetaMask/metamask-mobile/pull/4879/commits/1f420a374033f70aabe9efb273b6a1f75a55db24

Comment (blackdevelopa) [reply package.json]: 🤣🤣

**Review by blackdevelopa — APPROVED** [review]

**Review by cortisiko — APPROVED** [review]
This PR is QA passed. I tested:

- Sending tokens and NFTs to an on and off-chain ENS address
- I tested adding an off-chain ENS to my contacts
- I verified that https://github.com/MetaMask/metamask-mobile/pull/4318, https://github.com/MetaMask/metamask-mobile/pull/3825, and https://github.com/MetaMask/metamask-mobile/pull/4089 did not regress. 

Good work @makoto 

cc @gantunesr @mobularay

**Pull request merged**